packages/formation-react/src/components/DownloadLink/DownloadLink.jsx

```diff
@@ -0,0 +1,270 @@
+ * @return {ReactElement} renderedSize & renderedExtension wrapped in a `<dfn>`; or
+ *   customized as desired
+ */
+/**
+ * DownloadLink properties
+ * @typedef DownloadLink~props
+ * @type {object.<string, *>}
+ * @property {string} href (required) - url pointing to a downloadable file
+ * @property {string} title - title of the file; required if no children are passed
+ *  into this component
+ * @property {string|number} size - size of downloadable file; pass in string
+ *  (number + unit abbreviation, e.g. "1.5mb" or "100kb") or a file size number
+ *  in bytes.
+ * @property {string} download - new download file name and extension of the
+ *  downloadable file
+ * @property {string} type - downloadable file MIME-type; include if the file is
+ *  not in the list of known file extensions
+ * @property {string|ReactElement} icon - Fontawesome icon group + name, e.g.
+ *  "fas fa-download", or some custom icon JSX
+ * @property {boolean} external - Set to true if the link is to an external site;
+ *  setting this adds a blank target (opens a new tab) & no-referrer and opener
+ *  rel attribute
+ * @property {function} content - Function that allows custom rendering of the link
+ *  content; This function returns JSX
+ * @property {...*} var_args - Include any extra parameters you want to add to the
+ *  download link, e.g. `className`, `onClick`, etc.
+ */
+/**
+ * DownloadLink component
+ * @function DownloadLink
+ * @param {DownloadLink~props}
+ */
+const DownloadLink = props => {
+  if (!props.href) {
+    throw new Error('Download links require an href property');
+  }
+  const {
+    external = false,
+    title,
+    content,
+    download,
+    icon = '',
+    size,
+    ...linkProps
+  } = props;
+
+  if (!title && typeof content !== 'function') {
+    throw new Error(
+      'Download links require a title property or custom content function',
+    );
+  }
+
+  const fileName = getFileNameAndExt(linkProps.href);
+  const fileExt = getFileExtensionFromUrl(fileName);
+  const group = getExtGroup(fileExt);
+
+  if (external || !group) {
+    // Open external site in a new tab
+    linkProps.target = '_blank';
+    linkProps.rel = 'noopener noreferrer';
+  }
+
+  // fallback to `download` (no set value; filename stays the same)
+  linkProps.download = (download && processFileName(download)) || true;
+
+  if (!linkProps.type) {
+    linkProps.type =
+      (FILE_TYPES[group] && FILE_TYPES[group][fileExt]) ||
+      'application/unknown';
+  }
+
+  const renderedSize = renderFileSize(size);
+  const renderedExtension = fileExt && (
+    <abbr title={group || fileExt}>{fileExt.toUpperCase()}</abbr>
+  );
+  return (
+    <a {...linkProps}>
+      {typeof icon !== 'string' ? icon : createIcon(icon || 'fas fa-download')}
+      {(typeof content === 'function' &&
+        content({ renderedExtension, renderedSize, linkProps, props })) || (
+        <>
+          Download {title}{' '}
```

Comment:
Same question as with `MediaLink`; does it make sense to have default text in here?